compiler-rt/lib/asan/asan_errors.cpp

```diff
@@ -623,6 +662,10 @@ void ErrorGeneric::Print() {
     PrintContainerOverflowHint();
   ReportErrorSummary(bug_descr, &stack);
   PrintShadowMemoryForAddress(addr);
+
+  // This is an experimental flag, hence we don't make a special handler.
+  if (flags()->track_poison > 0)
+    CheckPoisonRecords(addr);
```

Comment:
Should we check it only if shadow matches use poison? or we can fetch stale data?

Reply:
I see, it's inside CheckPoisonRecords
maybe consider moving check for flags()->track_poison there as well?

Reply:
Moved